src/Core/Framework/Plugin/Command/PluginListCommand.php

```diff
@@ -83,33 +124,63 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 $pluginUpgradeable ? 'Yes' : 'No',
             ];
 
+            ++$this->total;
+
             if ($pluginActive) {
-                ++$active;
+                ++$this->active;
             }
 
             if ($pluginInstalled) {
-                ++$installed;
+                ++$this->installed;
             }
 
             if ($pluginUpgradeable) {
-                ++$upgradeable;
+                ++$this->upgradeable;
             }
         }
 
+        if ($format === 'json') {
+            $this->outputJson($io, $pluginTable);
+        } elseif ($format === 'table') {
+            $this->outputTable($io, $pluginTable, $filter);
+        }
+
+        return 0;
+    }
+
+    protected function outputTable(
+        ShopwareStyle $io,
+        array $pluginTable,
+        ?string $filter
+    ): void {
+        $io->title('Shopware Plugin Service');
+        if ($filter) {
+            $io->comment(sprintf('Filtering for: %s', $filter));
+        }
+
         $io->table(
             ['Plugin', 'Label', 'Version', 'Upgrade version', 'Author', 'Installed', 'Active', 'Upgradeable'],
             $pluginTable
         );
         $io->text(
             sprintf(
                 '%d plugins, %d installed, %d active , %d upgradeable',
-                \count($plugins),
-                $installed,
-                $active,
-                $upgradeable
+                $this->total,
+                $this->installed,
+                $this->active,
+                $this->upgradeable
             )
         );
+    }
 
-        return 0;
+    protected function outputJson(ShopwareStyle $io, array $pluginTable): void
+    {
+        $stats = [
+            'total' => $this->total,
+            'installed' => $this->installed,
+            'active' => $this->active,
+            'upgradeable' => $this->upgradeable,
+        ];
+        $io->write(json_encode(array_merge(['plugins' => $pluginTable, 'stats' => $stats])));
```

Comment:
array merge is not necessary here

Reply:
True, I first had two separate arrays there, forgot to remove it.